src/Compilers/CSharp/Test/Symbol/DocumentationComments/DocumentationCommentCompilerTests.cs

```diff
@@ -949,38 +965,521 @@ partial class C
+
+            var expectedDiagnostics = new[]
+            {
+                // (4,20): warning CS1570: XML comment has badly formed XML -- 'End tag 'a' does not match the start tag 'summary'.'
+                //     /// <summary></a></summary>
+                Diagnostic(ErrorCode.WRN_XMLParseError, "a").WithArguments("a", "summary").WithLocation(4, 20),
+                // (4,22): warning CS1570: XML comment has badly formed XML -- 'End tag was not expected at this location.'
+                //     /// <summary></a></summary>
+                Diagnostic(ErrorCode.WRN_XMLParseError, "<").WithLocation(4, 22)
+            };
+
+            // Files passed in order.
+            var compA = CreateCompilation(new[] { tree1, tree2 }, assemblyName: "Test");
+            compA.VerifyDiagnostics(expectedDiagnostics);
+            var actualA = GetDocumentationCommentText(compA);
+            var expected = @"
+<?xml version=""1.0""?>
+<doc>
+    <assembly>
+        <name>Test</name>
+    </assembly>
+    <members>
+        <!-- Badly formed XML comment ignored for member ""M:C.M"" -->
+    </members>
+</doc>
+".Trim();
+            AssertEx.Equal(expected, actualA);
+
+            // Files passed in reverse order.
+            var compB = CreateCompilation(new[] { tree2, tree1 }, assemblyName: "Test");
+            compB.VerifyDiagnostics(expectedDiagnostics);
+            var actualB = GetDocumentationCommentText(compB);
+            Assert.Equal(expected, actualB);
+        }
+
+        [Fact]
+        public void PartialMethod_Paramref_01()
+        {
+            var source1 = @"
+partial class C
+{
+    /** <summary>Accepts <paramref name=""p1""/>.</summary> */
+    public partial int M(int p1) => 42;
+}
+";
+
+            var source2 = @"
+partial class C
+{
+    public partial int M(int p2);
+}
+";
+
+            var tree1 = SyntaxFactory.ParseSyntaxTree(source1, options: TestOptions.RegularWithDocumentationComments);
+            var tree2 = SyntaxFactory.ParseSyntaxTree(source2, options: TestOptions.RegularWithDocumentationComments);
+
+            // Files passed in order.
+            verify(new[] { tree1, tree2 });
+
+            // Files passed in reverse order.
+            verify(new[] { tree2, tree1 });
+            
+            void verify(CSharpTestSource source)
+            {
+                var compilation = CreateCompilation(source, assemblyName: "Test");
+                compilation.VerifyDiagnostics(
+                    // (5,24): warning CS8826: Partial method declarations 'int C.M(int p2)' and 'int C.M(int p1)' have signature differences.
+                    //     public partial int M(int p1) => 42;
+                    Diagnostic(ErrorCode.WRN_PartialMethodTypeDifference, "M").WithArguments("int C.M(int p2)", "int C.M(int p1)").WithLocation(5, 24));
+
+                var method = compilation.GetMember<MethodSymbol>("C.M");
+                Assert.Equal("p2", method.Parameters.Single().Name);
```

Comment:
>Assert.Equal("p2", method.Parameters.Single().Name); [](http://example.com/codeflow?start=16&length=52)

I assume this assert is to verify what name is emitted. Rather than relying on the knowledge what will be emitted, we should verify what is emitted (CompileAnVerify + symbolVerifier).